manager/integration/tests/common.py

```diff
@@ -4027,7 +4034,9 @@ def wait_for_volume_restoration_start(client, volume_name, backup_name,
 @pytest.fixture
 def make_deployment_with_pvc(request):
     def _generate_deployment_with_pvc_manifest(deployment_name, pvc_name, replicas=1): # NOQA
-        make_deployment_with_pvc.deployment_manifest = {
+        if not hasattr(make_deployment_with_pvc, 'deployment_manifests'):
+            make_deployment_with_pvc.deployment_manifests = []
+        make_deployment_with_pvc.deployment_manifests.append({
```

Comment:
Why do you use a list here?

Reply:
It is in the finalizer so it get clean up: https://github.com/longhorn/longhorn-tests/blob/aa28f5d19238e0a5c483c7a5cbb9673c520ec651/manager/integration/tests/common.py#L4089-L4097

Reply:
Should this function just clean up the deployment it generates?

Reply:
> Should this function just clean up the deployment it generates?

Yeah, it should. With the fix, it will clean up the deployments that it created. Without the fix, it will only clean up the last deployment that it created

Reply:
This is a good catch!